sockeye/test_utils.py

```diff
@@ -58,6 +58,50 @@ def generate_digits_file(source_path: str,
             print(C.TOKEN_SEPARATOR.join(digits), file=target_out)
 
 
+def generate_json_input_file_with_tgt_prefix(src_path:str, tgt_path: str, json_file_with_tgt_prefix_path: str, \
+                                          src_factors_path: Optional[List[str]] = None, tgt_factors_path: List[str] = None, seed=13):
+    random_gen = random.Random(seed)
+    with open(src_path, "r") as src_reader, open(tgt_path, "r") as tgt_reader:
+        with open(json_file_with_tgt_prefix_path, "w") as out:
+            list_src_factors = None
+            list_tgt_factors = None
+
+            if src_factors_path is not None:
+                list_src_factors = [open(src_factors, "r") for src_factors in src_factors_path]
+                list_src_factors = [[sf.strip() for sf in src_factors] for src_factors in list_src_factors]
+
+            if tgt_factors_path is not None:
+                list_tgt_factors = [open(tgt_factors, "r") for tgt_factors in tgt_factors_path]
+                list_tgt_factors = [[tf.strip().split() for tf in tgt_factors] for tgt_factors in list_tgt_factors]
+
+            for i, stdigits in enumerate(zip(src_reader, tgt_reader)):
+                src_digits, tgt_digits = stdigits[0].strip(), stdigits[1].strip()
+                tgt_prefix = tgt_digits.split()
+                if len(tgt_digits) > 0:
+                    random_pos = random_gen.choice([pos for pos in range(len(tgt_prefix))])
```

Comment:
This can give you empty prefixes, right? What is the intended behavior then? Should we even allow this?

Reply:
This can give you empty prefixes, right? -> Yes
What is the intended behavior then? -> The intended behavior is to treat it like there is no prefix.
Should we even allow this? -> I don't know indeed Felix. I just thought: 

1 - it is very easy to handle that case, no need to adhoc modify anything.
2- No need for logger.error (Oops you wanted to use a prefix but the prefix is actually an empty string). 

Let me know if you have a different opinion.

Reply:
If empty prefixes work with the current implementation, thats great. But maybe we can add a logger.warning somewhere in the `TranslatorInput` construction code. A non-None prefix tensor will trigger your implementation in this PR to some extent. I don't think it would have a large effect on performance, but we should still warn the user, it may not be intended. What do you think?

Reply:
Done. You are right we should add warning for empty string of prefix token and empty list of prefix factors.